playground/src/email-loader/email-loader.js

```diff
@@ -21,31 +26,29 @@ export function createEmailLoader(editor) {
 class EmailLoader {
   constructor(editor) {
     this.editor = editor;
-  }
 
-  async loadEmailFromFile() {
-    const files = await new Promise((resolve) => {
-      const dialog = document.createElement('input');
-      dialog.setAttribute('type', 'file');
-      dialog.setAttribute('accept', '.eml');
-      dialog.addEventListener('change', () => resolve(dialog.files));
-      dialog.click();
+    events.subscribe(FileUpload.EVENT_FILE_UPLOADED, (file) => {
+      this.loadEmailContent(file);
     });
-    if (files.length !== 1) {
-      throw new Error('You must select a file');
+  }
+
+  async loadEmailContent(file) {
+    const data = await file.text();
+    try {
+      this.loadEmail(data);
+    } catch (error) {
+      events.publish(EVENT_FILE_UPLOADED_ERROR, `Error: ${error}`);
```

Comment:
I find `EVENT_FILE_UPLOADED_ERROR` to be misleading. I know the only case for a email to be loaded is by upload but if this event occurs there was actually a parsing error. So maybe `EVENT_LOAD_EMAIL_ERROR` or `EVENT_EML_PARSE_ERROR`, ...?